pbj-core/pbj-runtime/src/main/java/com/hedera/pbj/runtime/io/buffer/RandomAccessData.java

```diff
@@ -455,15 +455,10 @@ default String asUtf8String() {
      * @param offset the offset into the buffer to start reading bytes from
      * @param len the number of bytes to read
      * @return data converted to string
-     * @throws BufferUnderflowException if {@code len} is greater than {@link #length()} - {@code offset}.
      * @throws IndexOutOfBoundsException If the given {@code offset} is negative or not less than {@link #length()}
      */
     @NonNull
     default String asUtf8String(final long offset, final long len) {
-        if (len > length() - offset) {
-            throw new BufferUnderflowException();
-        }
-
         if (offset < 0 || offset + len > length()) {
```

Comment:
Why is it safe to remove this check? Can it never happen?

Reply:
`len > length() - offset` is the same as `offset + len > length()`, which is checked next.

Reply:
Oh, I see, it was moved to the subclass implementation in Bytes. But other subclasses of RandomAccessData may still want this check?

Reply:
Haha, indeed :-). Should we keep the first check and throw BufferUnderflowException in that case, and in the secon check just do offset < 0 for IndexOutOfBounds?

I don't have a real opinion here, but I think we should take a holistic view of which exceptions are thrown where to make sure they are consistent throughout all the API. Maybe that change should be done as a separate PR.

Reply:
The reason I got rid of BufferUnderflowException here is this exception is specific to buffers, when they are used to read data sequentially, but there is no data to read. This interface is `RandomAccessData`, which is more like an array than like a buffer, so IndexOutOfBoundsException makes much more sense to me